sdk/core/azure-core-amqp/inc/azure/core/amqp/message_sender.hpp

```diff
@@ -137,7 +137,7 @@ namespace Azure { namespace Core { namespace Amqp { namespace _internal {
      *
      * @return A tuple containing the status of the send operation and the send disposition.
      */
-    std::tuple<MessageSendStatus, Models::AmqpValue> Send(
+    std::tuple<MessageSendStatus, Models::_internal::AmqpError> Send(
```

Comment:
The AMQP value returned was always an AMQP Error wrapped in an AMQP List+AMQP described type. This just unwraps it to make the return consistent across all the AMQP message related APIs (management, message send, message receive).

When a message is sent, the receiver responds with a disposition indicating what was done with the message.

And the reason that AMQP returns values is to give the caller the ability to remediate errors coming from the various AMQP APIs - if you know the error reason that was returned, you can decide to reconnect and retry for example.

If it threw exceptions then the caller would have to catch them and then rewrite the caught exceptions (because we don't want AMQP exceptions to leak across the eventhubs abstraction layer). This way eventhubs generates the exception.